src/modeler/AutoRest.Swagger/Model/ServiceDefinitionMetadata.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace AutoRest.Swagger.Model
+{
+    /// <summary>
+    /// Describes whether a service definition has already been processed and merged.
+    /// This can influence ARM validation (e.g. one would not expect an operations API on individual OpenAPI definitions).
+    /// </summary>
+    public enum ServiceDefinitionMergeState
+    {
+        Before,
+        After
+    }
+
+    /// <summary>
+    /// Represents categories for OpenAPI definitions that are associated with special treatment with regards to validation.
+    /// </summary>
+    [Flags]
+    public enum ServiceDefinitionCategory
+    {
+        Regular = 0,
+        ARM = 1 << 0
```

Comment:
I like to do this for `Flags` style enums: For further categories, this will look like
```
... = 1 << 0,
... = 1 << 1,
... = 1 << 2,
... = 1 << 3,
... = 1 << 4,
... = 1 << 5
```
which I find just slightly more readable (and maybe less error prone if, say, someone decides to remove a category from the middle or something).

Kinda makes it clear that the categories actually *are* represented by bits ;-)